blender_imgui.py

```diff
@@ -338,18 +344,66 @@ def handler_remove(self, handle):
         if not self.callbacks:
             self.shutdown_imgui()
 
+    lastLetThroughTime = 0
+    def returnTrueXAmountOfTimesPerSecondToCreateSolidFramerate(self, framerate: int, frameNumber: list[int] = [0]):
+        """frame number should be something like: frameNumber = [0]. 
+        It will set index 0 of the list to a number that will always be incremented by 1 + any number of missed frames since last the function was called.
+
+        Example usecase of this function would be:
+
+        while(True):
+        frameNumber = [0]
+        if returnTrueXAmountOfTimesPerSecondToCreateSolidFramerate(4, frameNumber):
+            print(frameNumber)
+
+        which would print the current frame number 4 times every second. Each frame number would always be 1 greater than the previous."""
+        import time
+        import math
+        # framerate = 4
+        def formula():
+            return math.floor(time.time()*framerate)
+
+        currentTime = formula()
+        # try:
+        self.lastLetThroughTime
```

Comment:
Unused

Reply:
Hi, no its actually used at the beginning of GlobalImgui.draw at line 382.

This is method provides the "locking" mechanism that prevents imgui from making new frames above the max threshold provided by the global variable `blender_imgui_global_framerate_cap` at line 21